src/core/template-dx-integration.ts

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+
+import { generateID } from "./helpers";
+import { ITemplateUpdater } from "./template-updater";
+import { ITemplateWrapperProps, TemplateWrapper } from "./template-wrapper";
+
+interface ITemplateDxIntegration {
+    render: (data: IDxTemplateData) => any;
+}
+
+interface IDxTemplateData {
+    container: any;
+    model?: any;
+    index?: any;
+    onRendered?: () => void;
+}
+
+function createDxIntegration(
+    createContentProvider: () => (model: any) => any,
+    templateUpdater: ITemplateUpdater,
+    keyFn?: (data: any) => string
+): ITemplateDxIntegration {
+
+    const renderedContainers: HTMLElement[] = [];
+    return {
+        render: (data: IDxTemplateData) => {
+            const templateId = keyFn ? keyFn(data.model) : "__template_" + generateID();
+            const container = unwrapElement(data.container);
+
+            if (renderedContainers.indexOf(container) > -1) {
```

Comment:
Can we be sure the container is unique for each template result?

Reply:
Unfortunately, this works OK for the List, but is wrong in the case of DataGrid's rowTemplate (the same container is passed for all items - the underlying `<table/>` element)

Since this PR doesn't change behaviour, I will merge it and investigate the problem later